### ISSUE TITLE
Using "ui": "stream" results in turbo_json_parse_error "Found an unknown value `Stream`."

### ISSUE THREAD

**Issue by jckw** [vercel/turbo#8475]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

NA

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Other

### Which canary version will you have in your reproduction?

turbo@2.0.4-canary.3

### Describe the Bug

Turbo builds fail in Docker when the turbo.json "ui" value is set to "stream". 

This was previously reported in a thread last week (https://github.com/vercel/turbo/issues/8311#issuecomment-2149455716) and supposed to be fixed in 2.0.2 but it appears the bug is still there in Docker environments.

This error does not occur when building locally (on a Macbook Air M3).

Presumably this has something to do with the prune step or running turbo in Docker generally.

```
#24 0.290 yarn run v1.22.22
#24 0.310 $ /app/node_modules/.bin/turbo run build --filter=...web
#24 0.356 
#24 0.356 Attention:
#24 0.356 Turborepo now collects completely anonymous telemetry regarding usage.
#24 0.356 This information is used to shape the Turborepo roadmap and prioritize features.
#24 0.356 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
#24 0.356 https://turbo.build/repo/docs/telemetry
#24 0.356 
#24 0.356 turbo_json_parse_error
#24 0.356 
#24 0.356   x failed to parse turbo json
#24 0.356 
#24 0.356 Error: turbo_json_parse_error
#24 0.356 
#24 0.356   x Found an unknown value `Stream`.
#24 0.356     ,-[71:1]
#24 0.356  71 |   },
#24 0.356  72 |   "ui": "Stream"
#24 0.357     :         ^^^^^^^^
#24 0.357  73 | }
#24 0.357     `----
```

### Expected Behavior

It should build normally, like it does outside of Docker.

### To Reproduce

turbo.json
```json
{
  "$schema": "https://turbo.build/schema.json",
  "ui": "stream",
  ...
}
```

Dockerfile
```Dockerfile
...
RUN turbo prune web --docker
...
yarn turbo run build --filter=...web
```

**Comment by chris-olszewski** [comment]
Can you provide a reproduction. I just quick checked and it looks like this was fixed in 2.0.2. 

Can you verify that global `turbo` in `RUN turbo prune web --docker` is the version you expect? (`RUN turbo --version -vv`)

**Comment by jckw** [comment]
Ah yep, you're right - the version used in Docker was my issue. For some reason, `RUN yarn global add turbo` was using 2.0.0.

Everything works as expected once I updated that line to `RUN yarn global add turbo@latest` (which uses 2.0.4)

Thank you for your help!
